src/eddymotion/estimator.py

```diff
@@ -105,45 +107,40 @@ def fit(
                         # generate a synthetic dw volume for the test gradient
                         predicted = dwmodel.predict(data_test[1])
 
-                        tmpdir = Path(tmpdir)
-                        moving = tmpdir / "moving.nii.gz"
-                        fixed = tmpdir / "fixed.nii.gz"
-                        _to_nifti(data_test[0], dwdata.affine, moving)
-                        _to_nifti(
-                            predicted,
-                            dwdata.affine,
-                            fixed,
-                            clip=reg_target_type == "dwi",
-                        )
+                        moving_nii = nb.Nifti1Image(data_test[0], dwdata.affine)
+                        moving_ants = ants.from_nibabel(moving_nii)
+
+                        fixed_nii = nb.Nifti1Image(predicted, dwdata.affine)
+                        fixed_ants = ants.from_nibabel(fixed_nii)
+
+                        #registration_kwargs = json.loads('config/dwi-to-dwi_level1.json')
+                        #registration_kwargs['moving'] = moving_ants
+                        #registration_kwargs['fixed'] = fixed_ants
+                        registration_kwargs = dict(fixed=fixed_ants, moving=moving_ants)
 
-                        registration = Registration(
-                            terminal_output="file",
-                            from_file=pkg_fn(
-                                "eddymotion",
-                                f"config/dwi-to-{reg_target_type}_level{i_iter}.json",
-                            ),
-                            fixed_image=str(fixed.absolute()),
-                            moving_image=str(moving.absolute()),
-                            **align_kwargs,
-                        )
                         if bmask_img:
-                            registration.inputs.fixed_image_masks = ["NULL", bmask_img]
+                            registration_kwargs['mask'] = bmask_img
 
                         if dwdata.em_affines and dwdata.em_affines[i] is not None:
                             mat_file = tmpdir / f"init{i_iter}.mat"
                             dwdata.em_affines[i].to_filename(mat_file, fmt="itk")
-                            registration.inputs.initial_moving_transform = str(mat_file)
+                            registration_kwargs['initial_transform'] = str(mat_file)
+
+                        for key in registration_kwargs:
+                            registration_kwargs[key] = antsimage_from_path(registration_kwargs[key])
 
-                        # execute ants command line
-                        result = registration.run(cwd=str(tmpdir)).outputs
+                        #registration_kwargs = ants_utils._int_antsProcessArguments(registration_kwargs)
+                        registration_ants = ants.registration(**registration_kwargs)
```

Comment:
```suggestion
                        ants_cli = ants.registration(**registration_kwargs).cmline
                        
                        # Now split ants_cli and replace paths with image objects
                        # this is a bit more convoluted because some paths will come between brackets and commas
                        args = [load_if_image(token) for token in ants_cli.split(" ")]
```

Reply:
Could you show us where the .cmline is coming from? Are you suggesting that we revert back to the nipype Registration() call instead of the current ants call?